R/single_value.R

```diff
@@ -0,0 +1,29 @@
+#' Ensure that a vector has only a single value throughout.
+#'
+#' Missing values are replaced with the single value, and if all values are
+#' missing, the first value in \code{missing} is used throughout.
+#'
+#' @param x The vector which should have a single value
+#' @param missing The vector of values to consider missing in \code{x}
+#' @param warn_if_all_missing Generate a warning if all values are missing?
+#' @param info If more than one value is found, append this to the warning or
+#'   error to assist with determining the location of the issue.
+#' @return \code{x} as the scalar single value found throughout (or an error if
+#'   more than one value is found).
+#' @examples
+#' single_value(c(NA, 1))
+#' @export
+single_value <- function(x, missing=NA, warn_if_all_missing=FALSE, info=NULL) {
```

Comment:
This might be odd style but what do you think about `missing=c(NA)` as the default argument?  Then it hints to the user that a custom `missing` can take multiple values.

Reply:
I'll add that as an example, instead.  That will hopefully make it clearer (and I try to avoid the odd style... 😄 ).